build.assets/Dockerfile

```diff
@@ -142,9 +142,10 @@ RUN mkdir -p /opt && cd /opt && curl -L https://github.com/gravitational/libbpf/
 RUN go install github.com/google/addlicense@v1.0.0
 
 # Install golangci-lint.
-RUN (curl -L https://github.com/golangci/golangci-lint/releases/download/v1.46.0/golangci-lint-1.46.0-$(go env GOOS)-$(go env GOARCH).tar.gz | tar -xz && \
-     cp golangci-lint-1.46.0-$(go env GOOS)-$(go env GOARCH)/golangci-lint /bin/ && \
-     rm -r golangci-lint*)
+RUN go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.50.1
```

Comment:
This is quite the version bump, won't it cause problems? Do we need 1.50.x for GCI?

I'm happy to bump the linter, but you might have to do some additional work because of it.

Reply:
Yes, older `golangci-lint` still returns errors after running `gci`. I'll create a separate PR to update it.

Reply:
I ended up reverting the `golangci-lint` and downgrading the GCI to 0.7.0, so goimports linter doesn't complain about GCI changes. Seems to work fine.